examples/pytorch/hetgnn/main.py

```diff
@@ -1,90 +1,663 @@
+
+        for i, context in enumerate(data_loader):
+            optimizer.zero_grad()
+            loss, score, label = model(context)
+            scores.append(score)
+            labels.append(label)
+            loss.backward()
+            optimizer.step()
+
+            if i % 20 == 0:
+                end_time = time.time()
+                logger.info(
+                    "step: {:04d}; loss: {:.4f}; time: {:.4f}s".format(
+                        i, loss.data.item(), (end_time - start_time)
+                    )
+                )
+
+                times.append(end_time - start_time)
+                start_time = time.time()
+        scores = torch.unsqueeze(torch.cat(scores, 0), 1)
+        labels = torch.unsqueeze(torch.cat(labels, 0), 1).type(scores.dtype)
+        logger.info("Mean epoch MRR: {:.4f}".format(mrr(scores, labels)))
+
+    return model
+
+
+def save_embedding(model, graph, config):
+    model.eval()
+    batch_size = 200
+    saving_dataset = MockHetGnnFileNodeLoader(
+        graph=graph,
+        batch_size=batch_size,
+        model=model,
+        sample_file=config["sample_file"],
+    )
+
+    data_loader = torch.utils.data.DataLoader(saving_dataset)
+    with open(config["data_dir"] + "/node_embedding.txt", "w") as embed_file:
+        for _, context in enumerate(data_loader):
+            out_temp = model.get_embedding(context)
+            out_temp = out_temp.data.cpu().numpy()
+            inputs = context["inputs"].squeeze(0)
+            for k in range(len(out_temp)):
+                embed_file.write(
+                    str(inputs[k].numpy())
+                    + " "
+                    + " ".join([str(out_temp[k][x]) for x in range(len(out_temp[k]))])
+                    + "\n"
+                )
+
+
+def evaluate_link_prediction_on_het_gnn(input_data_map, config):  # noqa: F811
+    random.seed(0)
+    train_num, test_num = a_a_collaborate_train_test(config, input_data_map)
+    auc, f1 = evaluation.evaluate_link_prediction(config, train_num, test_num)
+    logger.info("Link prediction AUC: {:.4f} F1 score:{:.4f}".format(auc, f1))
 
-def _main():
-    # setup default logging component.
-    setup_default_logging_config(enable_telemetry=True)
+    assert auc > 0.73 and auc < 0.75
+    assert f1 > 0.66 and f1 < 0.69
 
-    # run_dist is the unified entry for pytorch model distributed training/evaluation/inference.
-    # User only needs to prepare initializing function for model, dataset, optimizer and args.
-    # reference: `deepgnn/pytorch/training/factory.py`
-    run_dist(
-        init_model_fn=create_model,
-        init_dataset_fn=create_dataset,
-        init_optimizer_fn=create_optimizer,
-        init_args_fn=init_args,
+
+def evaluate_classification_on_het_gnn(config):  # noqa: F811
+    random.seed(0)
+    train_num, test_num = a_class_cluster_feature_setting(config)
+    macroF1, microF1 = evaluation.evaluate_node_classification(
+        train_num, test_num, config
+    )
+
```

Comment:
This file is almost 1k lines, I think some stuff is not used can be removed.

Reply:
I think all of it is used. The main goal of this example is to show how to work with a pure python mock data.

Reply:
Can you refactor into multiple files then, I dont think huge files are very useful.

Reply:
This file has 625 lines, 1k is a brave round up.
I don't see any reference to file size in [PEP8](https://peps.python.org/pep-0008/)